lib/x86/adler32_impl.h

```diff
@@ -29,6 +29,17 @@
 #define LIB_X86_ADLER32_IMPL_H
 
 #include "cpu_features.h"
+#ifdef __INTEL_COMPILER
+typedef unsigned int __v4su __attribute__ ((__vector_size__ (16)));
```

Comment:
There is already a code block in `common/compiler_gcc.h` that defines missing vector types for gcc and clang when needed.  Could you put this in the same place, and reuse code if possible?

Reply:
done